compiler/x/codegen/OMRX86Instruction.cpp

```diff
@@ -1137,7 +1137,8 @@ void
 TR::X86RegImmSymInstruction::autoSetReloKind()
    {
    TR::Symbol *symbol = getSymbolReference()->getSymbol();
-   if (symbol->isConst())
+   if (symbol->isConst() ||
+       symbol->isConstantPoolAddress())
```

Comment:
Since these are OpenJ9 queries, shouldn't they be guarded with `J9_PROJECT_SPECIFIC` (as much as it pains me to add more of those)?

This is what's causing the CI failures.

Reply:
Oh wait, this requires #3324 ?

Reply:
I would note that the AOT infrastructure has not been factored for OMR and OpenJ9 yet (and is missing extension points etc). I agree J9 code should go in J9, imposing the complete re-architecture as an overhead to contribution is impractical given the scope of the code being discussed here - the refactor is a several large items itself.